xmppserver/src/main/java/org/jivesoftware/openfire/session/Session.java

```diff
@@ -47,9 +46,30 @@ public interface Session extends RoutableChannelHandler {
     int MAJOR_VERSION = 1;
     int MINOR_VERSION = 0;
 
-    int STATUS_CLOSED = -1;
-    int STATUS_CONNECTED = 1;
-    int STATUS_AUTHENTICATED = 3;
+    enum Status {
+        CLOSED,
+        CONNECTED,
+        AUTHENTICATED
+    }
+
+    /**
+     * @deprecated Use {@link Status#CLOSED} instead.
+     */
+    @Deprecated // Remove in or after Openfire 4.9.
+    Status STATUS_CLOSED = Status.CLOSED;
```

Comment:
You've marked this as deprecated, but by changing get/setStatus() you've already broken compatibility - is it worth keeping this at all?

Reply:
I had a plan for this (not sure if it holds up though): Those fields (constants really) initially were integers, but were refactored to enum values.

The `get/setStatus()` methods are changed, as you mentioned, but in the same way: they used integer values before, but now use the same enum values.

My reasoning for keeping the fields in (but marking them as being deprecated) is that I suspect that 99% of the usages will be combined with usage of the getter and setter. By changing the type of the fields _and_ the getter/setter to the same new types, compilation should not need to break (at least not when they're used together, like `if (session.getStatus() == Session.STATUS_CLOSED) { ... }` ).